packages/wrangler/src/deploy/index.ts

```diff
@@ -361,6 +363,15 @@ export async function deployHandler(
 		experimentalVersions: args.experimentalVersions,
 	});
 
+	if (deploymentId) {
+		writeOutput({
+			type: "deployment",
+			version: 1,
+			worker_id: name,
```

Comment:
Something I was just chatting with the team about - It would be a lot better for us to get a stable ID for the worker. Ideally we'd use the `script_tag` - Would that be possible?

Reply:
The `script_tag` is internal to EWC and not made available to Wrangler.

Reply:
All the APIs that Wrangler hits use the Worker name and not the ID. AFAIK.

Reply:
Wrangler does use script_tag for fetching deployments https://github.com/cloudflare/workers-sdk/blob/362bf3482da81d28d4e8821535e9b09319ceabe2/packages/wrangler/src/deployments.ts#L71 The problem with using `name` for us is that users can update the name in dash and then all of our internal references would be out-of-sync

Reply:
So basically someone has to make this call:

```
fetchResult<ServiceMetadataRes>(`/accounts/${accountId}/workers/services/${scriptName}`)).default_environment.script.tag;
```

I guess that Wrangler could do it since it will also have the authentication tokens etc.

I'll update to get that.

Reply:
deploy() already makes that call https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/deploy/deploy.ts so you can add it to the returned object

Reply:
Thanks @RamIdeas